### ISSUE TITLE
Feature Request: Support flymake

### ISSUE THREAD

**Issue by jgarte** [nemethf/eglot-x#4]
Hi,

Are there any plans to support the built-in flymake over flycheck?

**Comment by nemethf** [comment]
How does eglot-x support flycheck?  What do you mean by supporting flymake?  (The source code does not contain the string "fly".)

**Comment by jgarte** [comment]
>  (The source code does not contain the string "fly".)

Hi, what do you mean by this statement?

Technically the source code does contain the string "fly" exactly 20 times as of the current commit:

https://github.com/nemethf/eglot-x/blob/master/eglot-x.el#L1260

> How does eglot-x support flycheck?

Are you trying to point me towards the fact that you're making rpc calls like here and that support is being handled by rust-analyzer?

https://github.com/nemethf/eglot-x/blob/master/eglot-x.el#L1260

>  What do you mean by supporting flymake? 

I mean in the most general sense, that whatever functionalities work with flycheck, that if I enable flymake instead, they work as well.

Note that I'm not an Emacs, eglot, and LSP domain expert so If I am conflating anything, any explanations in correcting my mental model would be appreciated.

**Comment by nemethf** [comment]
> > (The source code does not contain the string "fly".)
> 
> Hi @nemethf , what do you mean by this statement?

Oh!  I made a series of mistakes :(.  First, I forgot I implemented *this* flycheck support.  Second, I was on a wrong branch, which did not contain this.  Sorry.
 
> > How does eglot-x support flycheck?
> 
> Are you trying to point me towards the fact that you're making rpc calls and the flycheck support is being handled by rust-analyzer?

So, this is has nothing to do with https://github.com/flycheck/flycheck (which is a flymake alternative that I never used.)

Eglot-x's flycheck commands  support rust-analyzer's [flycheck LSP extension](https://github.com/rust-lang/rust-analyzer/blob/master/docs/dev/lsp-extensions.md#controlling-flycheck).   I think rust-analyzer uses the "fly" prefix for automatic actions.  The rust-analyzer.checkOnSave config variable adjust when RA runs a cargo check.  Or the user can manually trigger a check with eglot-x-run-flycheck.  Maybe the docstring of this command is not clear on its own.  But hopefully the README helps a bit.

> > What do you mean by supporting flymake?
> 
> I mean in the most general sense, that whatever functionalities work with flycheck, that if I enable flymake instead, they work as well.

I think that's already true.
